src/realm/sync/instruction_replication.cpp

```diff
@@ -493,6 +497,21 @@ void SyncReplication::set(const Table* table, ColKey col, ObjKey key, Mixed valu
     }
 
     if (select_table(*table)) {
+        // Elide emission of Update(NULL, default=true) for embedded object /
+        // dictionary columns if the value is already NULL. This is a workaround
+        // for the fact that erase always wins for nested structures, but we
+        // don't want default values to win over later embedded object creation.
+        if (variant == _impl::instr_SetDefault && value.is_null()) {
```

Comment:
The IsDefault flag is intended for the SDK use case where the user has defined a default value in their object model (an unpersisted part of the schema / object model), but we don't want default-initialized fields to win over real field updates. So it's fine if the server never generates this kind of update, because this notion doesn't exist on the server.

Reply:
Right, that makes sense. The translator will always reject MongoDB documents that are missing required fields and not encode them at all, making `IsDefault` not necessary.